src/hb-ot-layout-gsubgpos.hh

```diff
@@ -3405,6 +3418,42 @@ struct GSUBGPOS
           && !alternate_feature_indices.has (i))
         feature_indices->del (i);
     }
+
+    //find out duplicate features after subset
+    unsigned prev = 0xFFFFu;
+    for (unsigned i : feature_indices->iter ())
+    {
+      if (prev == 0xFFFFu)
+      { prev = i; continue;}
+
+      hb_tag_t t = get_feature_tag (i);
+      hb_tag_t prev_t = get_feature_tag (prev);
+      if (t != prev_t)
+      { prev = i; continue; }
+
+      const Feature& f = get_feature (i);
+      const Feature& prev_f = get_feature (prev);
+
+      auto f_iter =
+      + hb_iter (f.lookupIndex)
+      | hb_filter (lookup_indices)
+      ;
+
+      auto prev_iter =
+      + hb_iter (prev_f.lookupIndex)
+      | hb_filter (lookup_indices)
+      ;
+
+      if (f_iter.len () != prev_iter.len ())
+      { prev = i; continue; }
+
+      bool is_equal = true;
+      for (auto _ : + hb_zip (f_iter, prev_iter))
+        if (_.first != _.second) { is_equal = false; break; }
+
+      if (is_equal == true) duplicate_index_map->set (i, prev);
+      else prev = i;
```

Comment:
This is only check for adjacent features that are duplicates. Is that done to match fontTools behaviour? Otherwise I would expect this to check each feature against all other features for duplicates.

Reply:
According to the spec, FeatureRecords are listed alphabetically by feature tag. See also: https://github.com/fonttools/fonttools/blob/9c0bc9e554766fe9755f070d1c048c8866922f0e/Lib/fontTools/feaLib/builder.py#L699

So checking for adjacent featureRecords should be enough.

Reply:
Ok cool, that makes sense then.